src/linker/Linker.Steps/SubStepsDispatcher.cs

```diff
@@ -0,0 +1,219 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+
+using Mono.Cecil;
+using Mono.Collections.Generic;
+
+namespace Mono.Linker.Steps
+{
+	//
+	// Generic steps dispatcher is intended to by used by custom linker step which
+	// consist of multiple steps. It simplifies their implementation as well as the
+	// way how to hook them into the pipeline of existing steps.
+	//
+	public abstract class SubStepsDispatcher : IStep, IEnumerable<ISubStep>
```

Comment:
Why is this abstract? It seems unnecessary to have to override this - I understand that custom steps will want to do that, so that they can register it as a custom step, but in general it feels unnecessary.

Reply:
Removed

Reply:
OK - I now realized that even inside the linker (without custom steps) the type names of steps are used to identify them. So it makes sense.